ledger/src/blockstore_processor.rs

```diff
@@ -971,7 +1031,18 @@ fn confirm_slot_entries(
     let slot = bank.slot();
     let (entries, num_shreds, slot_full) = slot_entries_load_result;
     let num_entries = entries.len();
-    let num_txs = entries.iter().map(|e| e.transactions.len()).sum::<usize>();
+    let mut entry_indexes = Vec::with_capacity(num_entries);
+    let mut entry_starting_index = progress.num_txs;
+    let num_txs = entries
+        .iter()
+        .map(|e| {
+            let num_txs = e.transactions.len();
+            let next_starting_index = entry_starting_index.saturating_add(num_txs);
+            entry_indexes.push((entry_starting_index..next_starting_index).collect::<Vec<_>>());
+            entry_starting_index = next_starting_index;
+            num_txs
+        })
+        .sum::<usize>();
```

Comment:
Rather than passing another `entry_indexes` vector as an extra argument that then has to be zipped up with the `entries` anyways, I think we should make a new `ReplayEntry` struct that contains:

1. The entry
2. The entry index